### PR TITLE
Less code by using more feature of argparse

### PR DESCRIPTION
**Pull request by GreatBahram**
HI @bbelderbos,

I added `ArgumentDefaultsHelpFormatter` as it helps to see the default values. Also, I saved the output of different ways you get the code (env, clipboard, file) into one attribute (code) and move the code inside get_code into argparse' type keyword, which allows execution of a custom function.

I also changed this line:
`CHROMEDRIVER_PATH = os.environ.get("CHROMEDRIVER_PATH", "")`
Because for someone like me who does not have the chrome driver, it throws an exception, probably it is a good idea to handle these edges and print a user-friendly error message. Maybe it is a good idea to move the code inside the `__main__` module to `cli`, I mean the `main` function because I think `cli` should be the main command-line endpoint and `__main__ ` should call the `cli.main` function, not vice versa (but maybe it is just matter of taste).

Thanks for your wonderful project

### CODE REVIEW

**Comment by bbelderbos** [comment]
Nice @GreatBahram looks great. Feel free to add the user-friendly message if `CHROMEDRIVER_PATH` is not set. Also could you check the tests? Seems they break because you now need to do `create_code_image(ONE_LINE_SNIPPET, **options)` explicitly for them to pass (affects all tests atm).

Thanks
Bob

File: carbon/cli.py [hunk 1]
```diff
@@ -3,49 +3,56 @@
...
-        "-c", "--clipboard", action="store_true", help="Use code on clipboard"
+        "-f", "--file", type=read_frm_file, help="File with code", dest="code"
     )
-    group.add_argument("-s", "--snippet", type=str, help="Code snippet")
     group.add_argument(
-        "-v", "--version", action="store_true", default=False, help="Show version"
+        "-c",
+        "--clipboard",
+        help="Use code on clipboard",
+        action="store_const",
```
Comment (bbelderbos) [inline carbon/cli.py:32]: nice, good to know

File: carbon/cli.py [hunk 2]
```diff
@@ -3,49 +3,56 @@
 
 import pyperclip
 
+from . import __version__ as version
+
+
+def read_frm_file(filename: str) -> str:
```
Comment (bbelderbos) [inline carbon/cli.py:9]: I would do `get_code_from_file`

File: carbon/cli.py [hunk 3]
```diff
@@ -3,49 +3,56 @@
...
+
 
 def get_args():
-    parser = argparse.ArgumentParser(description="Create a carbon code image")
+    parser = argparse.ArgumentParser(
+        description="Create a carbon code image",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "-v", "--version", action="version", version=f"%(prog)s {version}"
```
Comment (bbelderbos) [inline carbon/cli.py:22]: nice https://stackoverflow.com/a/15406624
did not know this, good to know

File: carbon/carbon.py [hunk 4]
```diff
@@ -9,20 +9,15 @@
...
-DEFAULT_LANGUAGE = "python"
-DEFAULT_BACKGROUND = "#ABB8C3"
-DEFAULT_THEME = "seti"
-
-
 def _create_carbon_url(code, **kwargs: str) -> str:
-    language = kwargs.get("language") or DEFAULT_LANGUAGE
-    background = kwargs.get("background") or DEFAULT_BACKGROUND
-    theme = kwargs.get("theme") or DEFAULT_THEME
+    language = kwargs.get("language")
```
Comment (bbelderbos) [inline carbon/carbon.py:18]: this change seems to break the tests, please check ...

Comment (GreatBahram) [reply carbon/carbon.py:18]: I'll fix it; thanks for mentioning it.

It seems like a good opportunity to work with GitHub actions to run the test cases.

Comment (GreatBahram) [reply carbon/carbon.py:18]: Hey Bod,

I added a fixture to pass the required keyword to solve the problem, sorry I couldn't download the chromedriver and test it but I think this must solve the problem

Comment (bbelderbos) [reply carbon/carbon.py:18]: Thanks @GreatBahram I still get 2 failures, but I will fix those quickly on main. Thanks for your contribution!

Comment (GreatBahram) [reply carbon/carbon.py:18]: Thank you very much, I really like this application

Take care,

Comment (bbelderbos) [reply carbon/carbon.py:18]: Thanks!

**Comment by bbelderbos** [comment]
Fixed tests. Another issue I hit:

```
$ make typing
mypy carbon tests
carbon/carbon.py:24: error: Argument 1 to "quote_plus" has incompatible type "Optional[str]"; expected "str"
carbon/carbon.py:26: error: Argument 1 to "quote_plus" has incompatible type "Optional[str]"; expected "str"
carbon/carbon.py:27: error: Argument 1 to "quote_plus" has incompatible type "Optional[str]"; expected "str"
Found 3 errors in 1 file (checked 6 source files)
```

So I made the options passed in required: https://github.com/PyBites-Open-Source/pybites-carbon/commit/f6f8d831722de5937a41142ec36f679748021269

**Comment by bbelderbos** [comment]
@GreatBahram Github Actions works now as well: https://github.com/PyBites-Open-Source/pybites-carbon/actions

**Comment by GreatBahram** [comment]
Sounds great, Bob! I really enjoyed that you used Tesseract to read characters from the generated images; that was a brilliant idea!

**Comment by bbelderbos** [comment]
Thanks, it doesn't work perfectly but enough to prove that the image with code was created!
I love finding creative ways to increase test coverage :)

**Pull request merged**